modules/islandora_audio/js/audio.js

```diff
@@ -4,44 +4,87 @@
  * @file
  * Displays Audio viewer.
  */
-(function ($, Drupal) {
-    'use strict';
+(function (Drupal, once) {
+  Drupal.behaviors.islandora_audio_captions = {
+    attach: function (context, settings) {
 
-    /**
-     * If initialized.
-     * @type {boolean}
-     */
-    var initialized;
-    /**
-     * Unique HTML id.
-     * @type {string}
-     */
-    var base;
+      once('islandora_audio_captions', 'audio', context).forEach(function (element) {
+        function parseTimestamp(timestamp) {
+          const [hours, minutes, seconds] = timestamp.split(':').map(Number);
+          return hours * 3600 + minutes * 60 + seconds;
+        }
+        function parseWebVTT(content) {
+          timeMarkerRegex = new RegExp("^\\d{2}:\\d{2}:\\d{2}\\.\\d{3} --> \\d{2}:\\d{2}:\\d{2}\\.\\d{3}$");
+          const lines = content.split('\n');
+          const cues = [];
+          let currentCue = null;
 
-    function init(context,settings){
-        if (!initialized){
-            initialized = true;
-            if ($('audio')[0].textTracks.length > 0) {
-                $('audio')[0].textTracks[0].oncuechange = function() {
-                  if (this.activeCues.length > 0) {
-                    var currentCue = this.activeCues[0].text;
-                    $('#audioTrack').html(currentCue);
-                  }
-                }
+          for (const line of lines) {
+            const trimmedLine = line.trim();
+            if (trimmedLine === 'WEBVTT') {
+              continue;
+            }
+            if (!trimmedLine) {
+              // Empty line indicates the end of a cue
+              if (currentCue) {
+                cues.push(currentCue);
+                currentCue = null;
+              }
+              continue;
             }
-        }
-    }
-    Drupal.Audio = Drupal.Audio || {};
 
-    /**
-     * Initialize the Audio Viewer.
-     */
-    Drupal.behaviors.Audio = {
-        attach: function (context, settings) {
-            init(context,settings);
-        },
-        detach: function () {
+            if (timeMarkerRegex.test(trimmedLine)) {
+              // Timing line
+              currentCue = { start: '', end: '', text: '' };
+              const [start, end] = trimmedLine.split(' --> ');
+              currentCue.start = parseTimestamp(start);
+              currentCue.end = parseTimestamp(end);
+            } else if (currentCue) {
+              // Text line
+              currentCue.text += (currentCue.text ? '\n' : '') + trimmedLine;
+            }
+          }
+          return cues;
```

Comment:
```suggestion
          if (typeof currentCue !== 'undefined') {
            cues.push(currentCue);
          }
          return cues;
```

I used the example linked in the PR comments. At 11:16 I'm not seeing

```
Arizona and you are so happily
matched.
```

The last text I see is

```
we all are who love our country, it is
a source of pride to me that
```

I think that last entry in the VTT needs added - something like in the suggestion.

Reply:
Thanks for catching that. I've added a commit.